tests/test_decompositions.py

```diff
@@ -49,6 +49,18 @@ def test_takagi_random_symm(self):
 
         self.assertAlmostEqual(error.mean() , 0)
 
+    def test_graph_embed_random_symm(self):
+        self.logTestName()
+        error=np.empty(nsamples)
+        max_mean_photon = 2
+        for i in range(nsamples):
+            X=random_degenerate_symmetric()
+            sc, U = dec.graph_embed(X, max_mean_photon=max_mean_photon)
+            error[i] = np.sinh(np.max(np.abs(sc)))**2 - max_mean_photon
+
+        self.assertAlmostEqual(error.mean() , 0)
```

Comment:
These tests seem a bit puzzling, and possibly strongly tied to the particular random seed.
What is the `sampled` part here? Is `graph_embed` stochastic?

Reply:
This test verifies that the maximum amount of squeezing used to encode the graph is indeed capped by the parameter max_mean_photon (the docstring has been updated to reflect this). There is only randomness in that we are checking this for random adjacency matrices - since we are looping over an ensemble and only asserting that the mean is equal to 0, the test is ~deterministic (using the same approach used across the SF test suite)

Reply:
could this test fail if the sample mean is, by chance, not within tolerance of zero?

Reply:
Updated this to add the argument `delta=self.tol`, so that it will use the default SF test suite tolerance.